src/rust/src/declarative_asn1/encode.rs

```diff
@@ -73,6 +73,20 @@ impl asn1::Asn1Writable for AnnotatedTypeObject<'_> {
                 let asn1_string: asn1::Utf8String<'_> = asn1::Utf8String::new(&val);
                 write_value(writer, &asn1_string)
             }
+            Type::PrintableString() => {
+                let val: &pyo3::Bound<'_, PrintableString> = value
+                    .downcast()
+                    .map_err(|_| asn1::WriteError::AllocationError)?;
+                let inner_str: pyo3::pybacked::PyBackedStr = val
+                    .get()
+                    .inner
+                    .extract(py)
+                    .map_err(|_| asn1::WriteError::AllocationError)?;
```

Comment:
We shouldn't need a pybacked string, `val.get().inner.bind(py)` should be all we need.